core/src/commonMain/kotlin/com/xebia/functional/xef/prompt/templates/templates.kt

```diff
@@ -5,13 +5,22 @@ import com.xebia.functional.xef.llm.models.chat.Role
 import com.xebia.functional.xef.prompt.Prompt
 import com.xebia.functional.xef.prompt.PromptBuilder
 import com.xebia.functional.xef.prompt.message
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.serializer
 
 fun system(context: String): Message = context.message(Role.SYSTEM)
 
 fun assistant(context: String): Message = context.message(Role.ASSISTANT)
 
 fun user(context: String): Message = context.message(Role.USER)
 
+inline fun <reified A> system(data: A): Message = system(Json.encodeToString(serializer(), data))
```

Comment:
Do you happen to know if this can be called from Java or Scala?

I feel like they should be,  but haven't checked. 

If you can confirm that,  just mark this as resolved.

Reply:
They can't be called from other JVM languages but we also expose a version that is not `inline` `reified` that takes in a serializer and in the JVM impl we can also take in a .class

Reply:
The `unaryPlus` and other methods meant to be used by kotlin are `@JvmSynthetic,` and they don't show up in scala and Java autocompletion or public members.